rest/config_legacy.go

```diff
@@ -246,6 +260,41 @@ func (dbc *DbConfig) ToDatabaseConfig() *DatabaseConfig {
 	}
 }
 
+func legacyServerAddressUpgrade(server string) (newServer, username, password string, err error) {
+	connSpec, err := connstr.Parse(server)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	// Get credentials from first host (if exists)
+	splitServers := strings.Split(server, ",")
+	u, err := url.Parse(splitServers[0])
+	if err != nil {
+		return "", "", "", err
+	}
+	if u.User != nil {
+		urlUsername := u.User.Username()
+		urlPassword, urlPasswordSet := u.User.Password()
+		if urlUsername != "" && urlPasswordSet {
+			username = urlUsername
+			password = urlPassword
+		}
+	}
+
+	if connSpec.Scheme == "http" {
+		connSpec.Scheme = "couchbase"
+		for i, addr := range connSpec.Addresses {
+			if addr.Port != 8091 {
```

Comment:
I thought we previously supported using http:// without specifying a port.  Does connstr.Parse fill in the default port if it's not specified?

Reply:
No it just sets the port as -1 (for not set)